strawberry/experimental/pydantic/fields.py

```diff
@@ -79,3 +83,42 @@ def get_basic_type(type_) -> Type[Any]:
         return new_type_supertype(type_)
 
     return type_
+
+
+def replace_pydantic_types(type_: Any, is_input: bool):
+    if isinstance(type_, type):
+        if issubclass(type_, BaseModel):
+            attr = "_strawberry_input_type" if is_input else "_strawberry_type"
+            if hasattr(type_, attr):
+                return getattr(type_, attr)
+            else:
+                raise UnregisteredTypeException(type_)
+    return type_
+
+
+def replace_types_recursively(type_: Any, is_input: bool) -> Any:
+    """Runs the conversions recursively into the arguments of generic types if any"""
+    basic_type = get_basic_type(type_)
+    replaced_type = replace_pydantic_types(basic_type, is_input)
+
+    origin = getattr(type_, "__origin__", None)
+    if origin is Literal:
+        # Literal does not have types in its __args__ so we return early
+        return replaced_type
+    if hasattr(replaced_type, "__args__"):
+        replaced_type = replaced_type.copy_with(
+            tuple(
+                replace_types_recursively(t, is_input=is_input)
+                for t in replaced_type.__args__
+            )
+        )
+        if isinstance(replaced_type, TypeDefinition):
+            # TODO: Not sure if this is necessary. No coverage in tests
+            # TODO: Unnecessary with StrawberryObject
+            replaced_type = builtins.type(
+                replaced_type.name,
+                (),
+                {"_type_definition": replaced_type},
+            )
```

Comment:
Can you add a test to cover this codepath?

Reply:
this was already present, i just moved the function :D